restlib/restlib.go

```diff
@@ -129,9 +134,14 @@ func DoHTTPRequest(ctx context.Context, client *http.Client, method string,
 	}
 
 	// OK
-	if httpResponse.StatusCode == http.StatusOK ||
-		httpResponse.StatusCode == http.StatusCreated ||
-		httpResponse.StatusCode == http.StatusAccepted {
+	switch httpResponse.StatusCode {
+	case http.StatusOK,
+		http.StatusCreated,
+		http.StatusAccepted,
+		http.StatusNonAuthoritativeInfo,
+		http.StatusNoContent,
+		http.StatusResetContent,
+		http.StatusPartialContent:
```

Comment:
I would remove `206 Partial Content` from the list. We never send a range header in the request and we always assume the full response is received.

Reply:
ok